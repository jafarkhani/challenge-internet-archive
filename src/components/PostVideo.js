

function PostVideo(props){

    return (
        <iframe 
            title={props.identifier}
            className="video-frame" 
            src={`https://archive.org/embed/${props.identifier}`}
            frameBorder="0" 
            webkitallowfullscreen="true" 
            mozallowfullscreen="true" 
            allowFullScreen ></iframe>
    );
}

export default PostVideo;