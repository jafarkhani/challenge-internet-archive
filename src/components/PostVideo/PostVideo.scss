
  .post-video-container{
    background-color: black;
  }
  
  .video-frame{
    width: 640px;
    height: 480px;
    max-width: 100%;
  }
  .post-video-features {
    border: 1px solid lightgray;
    color: lightgray;
    margin-left: 5px;
    padding: 15px;
    font-size: 300%;
    margin-right: 4px;
  }
  .post-video-star{ @extend .post-video-features; }
  .post-video-share{ @extend .post-video-features; }
  .post-video-flag{ @extend .post-video-features; }
  
  .post-video-star:hover{
    color : orange;
    border-color: orange;
  }
  
  .post-video-share:hover{
    color : blue;
    border-color: blue;
  }
  
  .post-video-flag:hover{
    color : red;
    border-color: red;
  }
  