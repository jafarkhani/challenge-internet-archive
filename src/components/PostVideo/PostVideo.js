import './PostVideo.scss';

function PostVideo(props){

    return (
        <iframe 
            data-testid="post-video"
            title={props.identifier}
            className="video-frame" 
            src={`https://archive.org/embed/${props.identifier}`}
            frameBorder="0" 
            webkitallowfullscreen="true" 
            mozallowfullscreen="true" 
            allowFullScreen ></iframe>
    );
}

export default PostVideo;