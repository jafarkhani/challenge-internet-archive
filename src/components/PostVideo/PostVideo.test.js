import React from "react";
import renderer from 'react-test-renderer';

import PostVideo from "./PostVideo";

describe("PostVideo", ()=> {

    it('snapshop Post Video', () => {
        const tree = renderer.create(<PostVideo identifier="test" />).toJSON();
        expect(tree).toMatchSnapshot();         
    });

})


