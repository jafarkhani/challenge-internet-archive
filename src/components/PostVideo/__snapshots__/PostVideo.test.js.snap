// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostVideo snapshop Post Video 1`] = `
<iframe
  allowFullScreen={true}
  className="video-frame"
  data-testid="post-video"
  frameBorder="0"
  mozallowfullscreen="true"
  src="https://archive.org/embed/test"
  title="test"
  webkitallowfullscreen="true"
/>
`;
